extends Node
class_name Level

func get_all_tile_map_layers():
    return find_child("TileMapLayerContainer",false,false).get_children()


func get_bound():
    var min_x = INF
    var min_y = INF
    var max_x = 0
    var max_y = 0
    for layer in get_all_tile_map_layers():
        var tile : TileMapLayer = layer
        var map_rect:Rect2i = tile.get_used_rect()
        var tile_size:Vector2i = tile.tile_set.tile_size
        var world_size := map_rect.size * tile_size
        var world_position := map_rect.position * tile_size
        min_x = min(min_x,world_position.x)
        min_y = min(min_y,world_position.y)
        max_x = max(max_x,world_position.x + world_size.x)
        max_y = max(max_y,world_position.y + world_size.y)
    return Rect2i(min_x,min_y, max_x - min_x, max_y-min_y)

# Called when the node enters the scene tree for the first time.
func _ready() -> void:
    pass # Replace with function body.


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta: float) -> void:
    pass
